lib/util/entrypoints.js

```diff
@@ -430,7 +430,24 @@ function targetMapping(
 		return mappingTarget + remainingRequest;
 	}
 	assert(mappingTarget, false);
-	return mappingTarget.replace(/\*/g, remainingRequest.replace(/\$/g, "$$"));
+	const wildcardIndex = mappingTarget.indexOf("*");
+
+	if (wildcardIndex !== -1) {
+		const maybeCommonSuffix = mappingTarget.slice(wildcardIndex + 1);
+		if (remainingRequest.endsWith(maybeCommonSuffix)) {
+			return mappingTarget.replace(
+				new RegExp(`\\*${maybeCommonSuffix}`, "g"),
```

Comment:
I think this needs to escape the characters that could have special meaning to the regex (like `.`)

Reply:
There is potential problem, we should handle only `*.js`, not `*` characters